cpu/atxmega/periph/spi.c

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (C) 2018 RWTH Aachen, Josua Arndt <jarndt@ias.rwth-aachen.de>
+ *               2021 Gerson Fernando Budke <nandojve@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_atxmega
+ * @ingroup     cpu_atxmega_periph
+ * @{
+ *
+ * @file
+ * @brief       Low-level SPI driver implementation
+ *
+ * @author      Josua Arndt <jarndt@ias.rwth-aachen.de>
+ * @author      Gerson Fernando Budke <nandojve@gmail.com>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "cpu_pm.h"
+#include "mutex.h"
+#include "periph/spi.h"
+#include "pm_layered.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+static inline void _print_buffer(const char* s, const uint8_t* buffer, uint16_t len)
```

Comment:
```suggestion
static void _print_buffer(const char* s, const uint8_t* buffer, uint16_t len)
```

When not in headers, we usually just mark functions as `static` and let the compiler do the optimization. I think compilers started to ignore the `inline` keyword some time ago and just do the inline-ing when it matches the optimization goals, and don't when not.